.github/src/changed-files.js

```diff
@@ -0,0 +1,114 @@
+// @ts-check
+
+import { diff } from "./git.js";
+
+/**
+ * @param {import('github-script').AsyncFunctionArguments['core']} core
+ * @param {string} [baseCommitish] Defaults to "HEAD^".
+ * @param {string} [headCommitish] Defaults to "HEAD".
+ * @returns {Promise<string[]>} List of changed files, relative to the repo root.  Example: ["specification/foo/Microsoft.Foo/main.tsp"]
+ */
+export async function getChangedFiles(
+  core,
+  baseCommitish = "HEAD^",
+  headCommitish = "HEAD",
+) {
+  // TODO: If we need to filter based on status, instead of passing an argument to `--diff-filter,
+  // consider using "--name-status" instead of "--name-only", and return an array of objects like
+  // { name: "/foo/baz.js", status: Status.Renamed, previousName: "/foo/bar.js"}.
+  // Then add filter functions to filter based on status.  This is more flexible and lets consumers
+  // filter based on status with a single call to `git diff`.
+  const result = await diff(baseCommitish, headCommitish, core, "--name-only");
+
+  const files = result.trim().split("\n");
+
+  core.info("Changed Files:");
+  for (const file of files) {
+    core.info(`  ${file}`);
+  }
+  core.info("");
+
+  return files;
+}
+
+// Functions suitable for passing to string[].filter(), ordered roughly in order of increasing specificity
+
+/**
+ * @param {string} [file]
+ * @returns {boolean}
+ */
+export function json(file) {
+  // Extension "json" with any case is a valid JSON file
+  return typeof file === "string" && file.toLowerCase().endsWith(".json");
```

Comment:
I'm fine with us using `toLowerCase` as the convention for case-insensitive string comparison for our purposes. I don't think `localeCompare` gives us any advantages and the semantics of `.toLowerCase().endsWith(...)` are much easier to reason about than something using `localeCompare`.

Reply:
Recommended way to do case-insensitive compare in JS is convert both strings to either lower or upper.  They don't provide a single `compareCaseInsensitive()` API.

I think we should default to case-sensitive filename comparison by default, but we may want to use case-insensitive for things like file extensions, where both "*.json" and "*.JSON" are generally considered "json files".